pkgs/tools/misc/man-db/default.nix

```
@@ -14,6 +14,9 @@
 , zstd
 }:
 
+let
+  useLibiconvReal = stdenv.hostPlatform.isDarwin || stdenv.hostPlatform.isFreeBSD;
```

Perhaps
```suggestion
  libiconv' = if stdenv.hostPlatform.isDarwin || stdenv.hostPlatform.isFreeBSD then libiconvReal else libiconv;
```
so that we don’t have to duplicate the logic in `buildInputs` and `nativeBuildInputs`. (Though of course the name isn’t particularly great.)

---

The way I like to do this is to name the attrset argument, then shadow the variable that's the most obvious.

```nix
useLibiconvReal = stdenv.hostPlatform.isDarwin || stdenv.hostPlatform.isFreeBSD;
libiconv = if useLibiconvReal then args.libiconvReal else args.libiconv;
```

---

I went with @emilazy’s suggested pattern. If the other is preferred, I can do that.